File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
         if(head == null){
            return null;
        }
        if(head.next == null){
            return head;
        }
        ListNode currH = head.next;
        ListNode curr = currH.next;
        head.next = null;
        currH.next = head;
        while(curr!=null){
            ListNode t = curr.next;
            ListNode x = currH;
            currH = curr;
            currH.next = x;
            curr = t;
        }
        return currH;
    }
}

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(!head) return null
    if(!head.next) return head
    let currH = head.next;
    let temp = currH.next;
    head.next = null;
    currH.next = head;
    while(temp){
        let curr = temp.next;
        temp.next = currH;
        currH = temp;
        temp = curr;
    }
    return currH
};